### PR TITLE
feat(auth): SubscriptionService extensibility (closes #24)

### LINKED ISSUE #24: Pericarp `SubscriptionService` extensibility
# Pericarp `SubscriptionService` extensibility — design spec

> Staged here for review. File as an issue on `akeemphilbert/pericarp` and link
> from apollo's epic #134; delete after upstream issue is closed and a new
> pericarp version is tagged.

## Job

Add a `SubscriptionService` interface to `pkg/auth` so subscription state can be looked up at token-issuance time and embedded as a JWT claim. Concrete implementations for RevenueCat, Stripe, and a generic GORM-backed adapter; consumer services read the claim off the validated token without per-request DB hops.

## Why this exists

Today pericarp's JWTs carry only identity claims (`agent_id`, `account_ids`, `active_account_id`). Services that gate functionality on a paid subscription (apollo's exercise generation, finexity's premium reports, future MCP tiers) end up bolting their own subscription lookup onto the request path:

- They re-implement the same "look up subscription by account_id, decide if active" logic in every receiver.
- They do that lookup on **every protected request** instead of once at token issuance.
- They standardize on a string sentinel ("inactive") in a custom JWT claim, which pericarp doesn't model — so apollo's exploration found `rest.SUBSCRIPTION_ID` references with no token-level support, which is exactly the gap that broke the apollo cutover (`SubscriptionFromContext` returns nil for everyone today).

Making subscription a first-class pericarp concept means: one canonical type, one place to integrate billing providers, one claim to read.

## Definition of done

- [ ] `pkg/auth/application` exposes `SubscriptionService` interface and a `SubscriptionClaim` value type.
- [ ] `AuthenticationService.IssueIdentityToken` calls the configured `SubscriptionService` (if any) and embeds the resulting claim in the issued JWT. No-op default when no service is configured (claim is omitted, matching today's behavior).
- [ ] `pkg/auth/application.PericarpClaims` gains an optional `Subscription *SubscriptionClaim` field.
- [ ] `pkg/auth/infrastructure/subscription/` ships at least three concrete implementations:
  - `revenuecat.go` — REST API client against `api.revenuecat.com/v1/subscribers/{agent_id}`
  - `stripe.go` — Stripe API client (`Customer.Subscriptions` lookup keyed by metadata or external_id)
  - `gorm.go` — generic adapter that reads from a service-owned `subscriptions` GORM table (matches apollo's existing `infrastructure.Subscription` projection)
- [ ] `WithSubscriptionService(svc)` functional option on `NewDefaultAuthenticationService`. When the option isn't used, `SubscriptionClaim` stays nil.
- [ ] Consumers can read the claim via `auth.AgentFromCtx(ctx).Subscription` after `BearerOrSession` (or any JWT-validating middleware) populates context.
- [ ] Tests: in-memory test exercising claim issuance with each adapter, plus a no-service fallback that confirms the JWT validates with the optional field absent.

## Anti-patterns

- **Don't fetch subscription on every request.** The whole point is moving the lookup off the hot path. The claim is set at `IssueIdentityToken` time and lives for the JWT's TTL (1h default). If a subscription state changes mid-token-lifetime, the user sees the new state on next refresh — that's the correct trade-off vs. a per-request billing API call.
- **Don't make `SubscriptionService` a pericarp-only interface that hides provider-specific fields.** Apollo needs `Status` (active/trialing/cancelled/past_due/inactive); other services may need `Plan` ("pro", "enterprise"); some services may need `Metadata` (stripe metadata, revenuecat entitlements). Keep the claim shape minimal and well-typed (`Status`, `Plan`, `ExpiresAt`, `Provider`); push provider-specific data into a `Metadata map[string]any` if needed.
- **Don't put plaintext billing identifiers (Stripe customer_id, RevenueCat app_user_id) in the JWT.** Those belong in pericarp's `account_members` extension or a separate billing-credential table, not in a token that crosses the wire.
- **Don't tie subscription state to `Account` lifecycle.** A user can have multiple accounts, each with separate subscriptions (B2B SaaS pattern). The claim is keyed on (`agent_id`, `active_account_id`), not on the agent alone.
- **Don't make the GORM adapter assume a specific schema.** Each consuming service has its own subscription projection (apollo has `infrastructure.Subscription`, finexity may differ). The GORM adapter should accept a query function or a small `Resolver` interface, not hard-code table names. Or alternatively, ship the GORM adapter as a *template* that services copy and parameterize.

## Out of scope

- Subscription mutation (creating, cancelling, refunding) — pericarp is read-only for billing state.
- Webhook handling for billing-provider state changes — that's the consumer service's job.
- Trial/grace-period business logic — just expose `Status` and `ExpiresAt`; the consumer interprets.
- Multi-tenancy of billing accounts — each agent has one active account at a time per session; that's pericarp's existing model.

### Domain / value type

```go
// pkg/auth/application/subscription_service.go
package application

import (
    "context"
    "time"
)

type SubscriptionStatus string

const (
    SubscriptionStatusActive    SubscriptionStatus = "active"
    SubscriptionStatusTrialing  SubscriptionStatus = "trialing"
    SubscriptionStatusPastDue   SubscriptionStatus = "past_due"
    SubscriptionStatusCancelled SubscriptionStatus = "cancelled"
    SubscriptionStatusInactive  SubscriptionStatus = "inactive"
)

type SubscriptionClaim struct {
    Status    SubscriptionStatus `json:"status"`
    Plan      string             `json:"plan,omitempty"`
    Provider  string             `json:"provider,omitempty"`
    ExpiresAt time.Time          `json:"expires_at,omitempty"`
}

// IsActive reports whether the subscription is in a state that should grant
// paid-tier access. Centralizes the "what counts as paying" rule so consumers
// don't drift into ad-hoc checks.
func (s *SubscriptionClaim) IsActive() bool {
    if s == nil { return false }
    switch s.Status {
    case SubscriptionStatusActive, SubscriptionStatusTrialing:
        return true
    }
    return false
}

type SubscriptionService interface {
    GetSubscription(ctx context.Context, agentID, accountID string) (*SubscriptionClaim, error)
}
```

### JWT claim integration

`PericarpClaims` gains:

```go
type PericarpClaims struct {
    jwt.RegisteredClaims
    AgentID         string             `json:"agent_id"`
    AccountIDs      []string           `json:"account_ids"`
    ActiveAccountID string             `json:"active_account_id,omitempty"`
    Subscription    *SubscriptionClaim `json:"subscription,omitempty"`
}
```

`IssueIdentityToken` (and `RSAJWTService.IssueToken`):

```go
if s.subscriptionService != nil {
    sub, err := s.subscriptionService.GetSubscription(ctx, agent.GetID(), activeAccountID)
    if err != nil {
        s.logger.Warnf("subscription lookup failed for %s: %v", agent.GetID(), err)
    } else {
        claims.Subscription = sub
    }
}
```

Lookup failures don't block token issuance — the user still authenticates, but the claim is absent (which the consumer interprets as "no active subscription"). This avoids billing-provider outages bringing down login.

### Functional option

```go
func WithSubscriptionService(svc SubscriptionService) AuthServiceOption {
    return func(s *DefaultAuthenticationService) {
        if svc != nil { s.subscriptionService = svc }
    }
}
```

### Concrete implementations

**RevenueCat** (`pkg/auth/infrastructure/subscription/revenuecat.go`)
- Constructor: `NewRevenueCat(apiKey string, opts ...Option) *RevenueCat`
- `GetSubscription` calls `GET https://api.revenuecat.com/v1/subscribers/{agent_id}`, parses entitlements, maps to `SubscriptionClaim`.
- Provider field set to `"revenuecat"`.

**Stripe** (`pkg/auth/infrastructure/subscription/stripe.go`)
- Constructor: `NewStripe(apiKey string, opts ...Option) *Stripe`
- `GetSubscription` looks up customer by `metadata[agent_id]` (or a configured key), reads active subscriptions, maps the latest active one to `SubscriptionClaim`.
- Provider field set to `"stripe"`.

**GORM** (`pkg/auth/infrastructure/subscription/gorm.go`)
- Generic adapter: takes a `Resolver func(ctx, agentID, accountID string) (*SubscriptionClaim, error)` so each service can define its own table query. Or simpler: takes a `*gorm.DB` plus a configurable struct/table name, expects columns `account_id`, `status`, `plan`, `expires_at`, `provider`.
- Used by services that own billing projections (apollo's case post-cutover).

### Configuration

Per-provider env vars, all optional:

| Env var | Used by | Purpose |
|---|---|---|
| `REVENUECAT_API_KEY` | `infrastructure/subscription/revenuecat.go` | RevenueCat REST auth |
| `STRIPE_API_KEY` | `infrastructure/subscription/stripe.go` | Stripe REST auth |
| `STRIPE_AGENT_METADATA_KEY` | same | metadata field name on Stripe customer (default: `"agent_id"`) |

Consumer services wire the provider that matches their billing stack:

```go
// apollo: GORM adapter against infrastructure.Subscription
fx.Provide(func(db *gorm.DB) authapp.SubscriptionService {
    return subscription.NewGORM(db, subscription.WithTable("subscriptions"))
})

// then in ProvideAuthenticationService:
return authapp.NewDefaultAuthenticationService(
    registry, agents, credentials, sessions, accounts,
    authapp.WithPasswordCredentialRepository(passwords),
    authapp.WithJWTService(jwt),
    authapp.WithSubscriptionService(subSvc),
)
```

## Validation

Tests that should pass once this lands:

```go
// no-op default
authService := authapp.NewDefaultAuthenticationService(registry, agents, credentials, sessions, accounts,
    authapp.WithJWTService(jwt))
token, _ := authService.IssueIdentityToken(ctx, agent, accountID)
claims, _ := jwt.ValidateToken(ctx, token)
require.Nil(t, claims.Subscription)

// with adapter
mockSvc := &fakeSubscriptionService{result: &authapp.SubscriptionClaim{
    Status: authapp.SubscriptionStatusActive,
    Plan: "pro",
    ExpiresAt: time.Now().Add(30 * 24 * time.Hour),
    Provider: "stripe",
}}
authService := authapp.NewDefaultAuthenticationService(registry, agents, credentials, sessions, accounts,
    authapp.WithJWTService(jwt),
    authapp.WithSubscriptionService(mockSvc))
token, _ := authService.IssueIdentityToken(ctx, agent, accountID)
claims, _ := jwt.ValidateToken(ctx, token)
require.NotNil(t, claims.Subscription)
require.True(t, claims.Subscription.IsActive())
require.Equal(t, "pro", claims.Subscription.Plan)
```

## Reference implementations

- Apollo's existing `infrastructure.Subscription` GORM projection — the GORM adapter should be a drop-in fit for it.
- Apollo's exercise paywall logic (`domain/exercise.go` `With`/`WithSimilarity`/`WithBespoke`/`FromExercise`) — null-checks `*Subscription`. Post-this-feature, the receivers populate `*Subscription` from `auth.AgentFromCtx(ctx).Subscription` instead of from `rest.SUBSCRIPTION_ID`.
- Stripe Customer.Subscriptions API: <https://docs.stripe.com/api/subscriptions/list>
- RevenueCat Subscribers API: <https://www.revenuecat.com/reference/subscribers>

### PR DESCRIPTION
**Pull request by akeemphilbert**
Closes #24.

All four stories are landed on this branch with /pr-review feedback addressed on each:

## Story 1 — Core interface + JWT integration
- `auth.SubscriptionClaim` value type + `SubscriptionStatus` constants + `IsActive()` (consults `ExpiresAt` so a stale snapshot can't grant access past the provider-attested expiry); `SubscriptionStatus.Valid()` to catch provider-string drift.
- `application.SubscriptionService` interface; `WithSubscriptionService` functional option; `PericarpClaims.Subscription` field.
- `JWTService.IssueToken` gains a 5th `*auth.SubscriptionClaim` param. `ReissueToken` preserves the claim verbatim.
- `AuthenticationService.IssueIdentityToken` orchestrates the lookup. **Lookup failures are logged but do NOT block token issuance** — billing-provider outages must not break login.
- `RequireJWT` populates `auth.Identity.Subscription`; `RequireAuth` (session) leaves it nil.

## Story 2 — RevenueCat adapter (`pkg/auth/infrastructure/subscription/revenuecat.go`)
- `GET /v1/subscribers/{agent_id}` → `*auth.SubscriptionClaim`. Maps the latest-expiring entitlement; lifetime entitlements (null `expires_date`) win over time-bounded.
- Honors `unsubscribe_detected_at` (refunded lifetime → Inactive); `billing_issues_detected_at` (PastDue); `period_type=trial` (Trialing).
- Falls back to scanning all subscriptions when the matched-row keying misses (multi-SKU offerings, upgrades, promo grants).
- agentID `url.PathEscape`d.

## Story 3 — Stripe adapter (`pkg/auth/infrastructure/subscription/stripe.go`)
- `customers/search?query=metadata['agent_id']:'...'&expand[]=data.subscriptions` → `*auth.SubscriptionClaim`. Configurable metadata key via `WithStripeAgentMetadataKey`.
- `canceled` with future `current_period_end` keeps `IsActive()` true (paid through, just won't renew) — surfaced via `cancel_at_period_end` metadata flag. Once the period lapses it becomes Cancelled.
- Selection ranks active/trialing > past_due > canceled; ties on later period_end then subscription ID for stable output.
- Plan precedence: `lookup_key` → `nickname` → `product`.
- Metadata stamps `customer_id` and `customer_match_count` (split-brain billing setups become detectable).
- Apostrophes in agentID backslash-escaped per Stripe's Search API grammar.

## Story 4 — GORM adapter (`pkg/auth/infrastructure/subscription/gorm.go`)
- Default schema: `SubscriptionRecord` (agent_id, account_id, status, plan, provider, expires_at, updated_at).
- **Strict account scoping**: when accountID is non-empty, requires an exact (agent_id, account_id) match — no agent-only fallback — so a paid personal-account subscription cannot silently grant paid-tier access to a B2B/team account the same agent belongs to.
- Latest `updated_at` wins; ties break on `id DESC` for determinism.
- `WithGORMTable` overrides the table name; `WithGORMResolver` is the fully-custom-query escape hatch; `WithGORMProvider` stamps a fallback Provider when the row column is empty (row value wins when present).
- Validates returned `Status` against `SubscriptionStatus.Valid()` — a typo'd "ACTIVE" or schema drift surfaces as a loud error rather than a silent token claim.
- Returns `(nil, nil)` for ErrRecordNotFound; propagates real DB failures.

## Test plan
- [x] `make dev-test` (fmt + lint + race-enabled tests) — clean
- [x] 69 tests in `pkg/auth/infrastructure/subscription` covering happy paths, status mappings, selection precedence, fallback semantics, error propagation, options nil-safety, and provider-specific edge cases (RevenueCat lifetime revocation, Stripe canceled-but-paid-window, GORM cross-account leak guard)
- [x] Round-trip tests at `pkg/auth/infrastructure/jwt` confirm `Status`, `Plan`, `Provider`, `ExpiresAt`, `Metadata` all survive sign/validate
- [x] Middleware test confirms `auth.Identity.Subscription` is populated end-to-end via JWT and stays nil under session-only auth
- [ ] Manual validation in apollo against the GORM adapter once cut over (deferred to consumer)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/auth/subscription.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+//
+// Metadata carries provider-specific fields (e.g., Stripe metadata,
+// RevenueCat entitlement keys). It is preserved verbatim so consumers can
+// reach into provider-native data when the normalized fields are not enough.
+type SubscriptionClaim struct {
+	Status    SubscriptionStatus `json:"status"`
+	Plan      string             `json:"plan,omitempty"`
+	Provider  string             `json:"provider,omitempty"`
+	ExpiresAt time.Time          `json:"expires_at,omitzero"`
+	Metadata  map[string]any     `json:"metadata,omitempty"`
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/subscription.go]: Addressed in 45e629b: kept `omitzero` (intentional choice for Go 1.24+) and added a doc comment on the field explaining why — `omitempty` would emit `"0001-01-01T00:00:00Z"` for unset times instead of dropping the field, which is the bug we're avoiding. The standard `encoding/json` in Go 1.24+ honors `omitzero`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/revenuecat.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	for _, opt := range opts {
+		opt(r)
+	}
+	return r
+}
+
+// revenueCatResponse is the trimmed shape of GET /v1/subscribers/{app_user_id}.
+// Only the fields the adapter consumes are decoded; the rest of RevenueCat's
+// response is preserved opaquely in SubscriptionClaim.Metadata via
+// raw_entitlements when the caller wants provider-native data.
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/revenuecat.go]: Addressed in 872fb9d — the misleading `raw_entitlements` claim was dropped from the doc comment on `revenueCatResponse`. It now reads "Only the fields the adapter consumes are decoded."

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/revenuecat.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	_ = accountID
+	if r.apiKey == "" {
+		return nil, errors.New("revenuecat: missing API key")
+	}
+	if agentID == "" {
+		return nil, errors.New("revenuecat: agentID must not be empty")
+	}
+
+	url := fmt.Sprintf("%s/subscribers/%s", r.baseURL, agentID)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/revenuecat.go]: Addressed in 872fb9d — `agentID` is now `url.PathEscape`'d before being interpolated into the subscribers URL (revenuecat.go:126).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .claude/journal.md [hunk 4]
```diff
@@ -41,6 +41,28 @@ tasks to maintain context across sessions. Entries are never edited or removed.
...
+- `AuthenticationService.IssueIdentityToken` orchestrates the lookup when a `SubscriptionService` is wired via `WithSubscriptionService(svc)`. Lookup failures are logged but do **not** block token issuance — billing-provider outages must not break login
+- `RequireJWT` middleware now copies `claims.Subscription` onto `auth.Identity.Subscription`, so consumer services read it via `auth.AgentFromCtx(ctx).Subscription`. Session-based `RequireAuth` leaves it nil (sessions don't snapshot subscription state)
+- **Why:** Apollo and other downstream services were hand-rolling subscription lookups on every protected request. Promoting subscription to a first-class JWT claim moves the lookup off the hot path (looked up once at issuance, lives for token TTL) and gives every consumer a single canonical type to read
+- **Open follow-ups (Stories 2-4):** RevenueCat, Stripe, and GORM adapter implementations under `pkg/auth/infrastructure/subscription/`
+
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply .claude/journal.md:74]: Addressed in 0fbf02c — a new journal entry "SubscriptionService adapters — RevenueCat / Stripe / GORM (Stories 2-4 of #24)" now documents the implementations as landed. The older Story 1 entry retains its "open follow-ups" line as a historical record of what was open at the time.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/stripe.go [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+	if s.apiKey == "" {
+		return nil, errors.New("stripe: missing API key")
+	}
+	if agentID == "" {
+		return nil, errors.New("stripe: agentID must not be empty")
+	}
+
+	// Stripe search syntax: metadata['key']:'value'. Single quotes inside
+	// agentID are escaped per Stripe's docs by doubling — same convention
+	// SQL uses. Limits the blast radius of a malformed agent ID.
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/stripe.go]: Addressed in 3811573 — the comment was corrected to describe the implemented strategy (apostrophes in agentID are backslash-escaped as `\'` per Stripe's Search API grammar).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/subscription_test.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		{"frobnicated", false},
+	}
+
+	for _, tc := range tests {
+		t.Run(string(tc.status), func(t *testing.T) {
+			t.Parallel()
+			if got := tc.status.Valid(); got != tc.valid {
+				t.Errorf("Valid(%q) = %v, want %v", tc.status, got, tc.valid)
+			}
+		})
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/subscription_test.go:67]: This is no longer required: as of Go 1.22, `for _, tc := range tests` creates a per-iteration variable, so the closure capture issue this rule was guarding against doesn't apply. The repo is on Go 1.25 (`go.mod`), so the parallel subtests are safe as written.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/subscription_test.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		{name: "empty status", claim: &SubscriptionClaim{}, active: false},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			if got := tc.claim.IsActive(); got != tc.active {
+				t.Errorf("IsActive() = %v, want %v", got, tc.active)
+			}
+		})
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/subscription_test.go:39]: Same as the other `tc` thread: Go 1.22+ gave `for _, tc := range` per-iteration loop variable semantics, and this repo is on Go 1.25, so the parallel-subtest closure capture concern doesn't apply.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/revenuecat.go [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+	PeriodType              string     `json:"period_type"`
+	Store                   string     `json:"store"`
+}
+
+// GetSubscription queries RevenueCat for agentID's current subscription
+// state. accountID is unused — RevenueCat keys on a single app_user_id, so
+// callers should arrange to use agentID consistently across registration
+// and lookup. Returns (nil, nil) when the subscriber has no active or
+// recently-active entitlement; returns an error for transport failures and
+// non-2xx responses other than 404 (which is also treated as "no record").
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/revenuecat.go]: Addressed in 45cccf8 — `GetSubscription`'s doc comment now states that `(nil, nil)` is only returned on 404 or no entitlement data, and that expired entitlements still produce a non-nil claim with inactive status.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/revenuecat.go [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+
+func readSubscriptionSignals(all map[string]revenueCatSubscription, matched revenueCatSubscription, hasMatch bool) subscriptionSignals {
+	var s subscriptionSignals
+	if hasMatch {
+		s.hasBillingIssues = matched.BillingIssuesDetectedAt != nil
+		s.isTrial = matched.PeriodType == "trial"
+		s.unsubscribedAt = matched.UnsubscribeDetectedAt
+		s.unsubscribedExpiresAt = matched.ExpiresDate
+		s.matchedHadUnsubscribe = matched.UnsubscribeDetectedAt != nil
+		return s
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/revenuecat.go:260]: Addressed in 45cccf8 — removed `matchedHadUnsubscribe` from `subscriptionSignals` and its assignment in `readSubscriptionSignals`. The status logic only consults `unsubscribedAt`/`unsubscribedExpiresAt`, which are still populated.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/gorm.go [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
+		if err != nil {
+			return nil, fmt.Errorf("gorm: query account-scoped subscription: %w", err)
+		}
+		return g.toClaim(rec), nil
+	}
+
+	err := base.Where("account_id = ? OR account_id IS NULL", "").
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/gorm.go]: Addressed in 45cccf8 — wrapped the OR clause in parentheses: `Where("(account_id = ? OR account_id IS NULL)", "")`. The earlier form risked SQL precedence widening the match to NULL `account_id` rows for any agent.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/gorm.go [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+)
+
+// SubscriptionRecord is the default row shape the GORM adapter reads from.
+// Consumer services that already own a billing projection can either
+// AutoMigrate this struct into their schema or override the table name via
+// WithGORMTable; for fully custom schemas use WithGORMResolver.
+type SubscriptionRecord struct {
+	ID        uint       `gorm:"primaryKey"`
+	AgentID   string     `gorm:"size:64;index:idx_sub_agent;not null"`
+	AccountID string     `gorm:"size:64;index:idx_sub_agent_account"`
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/gorm.go]: Addressed in 45cccf8 — both `AgentID` and `AccountID` now share `idx_sub_agent_account` with priorities 1 and 2, so AutoMigrate creates the composite `(agent_id, account_id)` index the queries actually need. Dropped the redundant single-column `idx_sub_agent` since `agent_id` is the leading column of the composite.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/application/authentication_service.go [hunk 12]
```diff
@@ -518,7 +523,16 @@ func (s *DefaultAuthenticationService) IssueIdentityToken(ctx context.Context, a
...
 		}
 	}
-	return s.jwtService.IssueToken(ctx, agent, accounts, activeAccountID)
+	var subscription *auth.SubscriptionClaim
+	if s.subscriptionService != nil {
+		sub, err := s.subscriptionService.GetSubscription(ctx, agent.GetID(), activeAccountID)
+		if err != nil {
+			s.logger.Warn(ctx, "auth: subscription lookup failed", "agent_id", agent.GetID(), "active_account_id", activeAccountID, "error", err)
+		} else {
+			subscription = sub
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/application/authentication_service.go]: Addressed in 07b49bb — `IssueIdentityToken` now validates `sub.Status.Valid()` before embedding the claim and logs a warning (with `agent_id`, `active_account_id`, `status`) when an adapter returns a non-canonical status string. Mirrors the boundary validation the GORM adapter already performs.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/auth/infrastructure/subscription/stripe.go [hunk 13]
```diff
@@ -0,0 +1,327 @@
...
+			claim.Metadata = map[string]any{}
+		}
+		claim.Metadata["subscription_id"] = best.ID
+	}
+	if bestCust != "" {
+		if claim.Metadata == nil {
+			claim.Metadata = map[string]any{}
+		}
+		claim.Metadata["customer_id"] = bestCust
+	}
```
[the remark this answers is not in the document]
Comment (akeemphilbert) [reply pkg/auth/infrastructure/subscription/stripe.go:269]: Addressed in 65bcac6 (pushed post-merge to `pericarp-24`) — `customer_id` and `subscription_id` are now omitted from `SubscriptionClaim.Metadata` by default. New `WithStripeIncludeBillingIDs(true)` option opts back in for callers that explicitly need JWT↔Stripe correlation. `customer_match_count` is unchanged because it's a count rather than an identifier, so split-brain detection still works without leaking IDs. Note: PR #25 was merged before this fix landed; the commit is on the merged branch and will need a follow-up PR to reach `main`.

---
_Generated by [Claude Code](https://claude.ai/code)_
